specs/consensus/overview.md

```diff
@@ -651,32 +651,133 @@ be implemented by the processes running the consensus protocol.
+What Property 2 defines is that if any correct process receives and processes
+`m`, then all correct processes must eventually receive `m` as well;
+the same applies to the conflicting message `m'`.
+This property is not trivial to ensure.
+
+> Property 1 can be ensured in a fully-connected network, where every pair of
+> processes is connected via a reliable channel.
+> In this case, if the sender of the message is correct, then every correct
+> destination eventually receives the message.
+>
+> In the same environment, ensuring Property 2 requires correct to relay the
+> received messages.
+> In this way, if the sender crashes and the message is not received by some
+> correct processes, the other correct processes will relay it to them.
+> In the same way, if the sender is Byzantine and purposely does not send the
+> message to a subset of processes, the message will be relayed to them.
+>
+> The complexity added by Property 2 is the need for a protocol for reliably
+> broadcasting messages in a Byzantine setup.
+> In other words, the `broadcast` primitive cannot be just implemented by
+> sending the same message (via "unicast") to every process.
+
+[comment]: <> (I wonder whether the next paragraph is more design related)
+
+Fortunately, it turns out that the Gossip communication property does not need
+to be ensured for _every_ broadcast message.
+As discussed in [#260][synchronization-issue], only messages whose reception
+produce a relevant action in the consensus state-machine (in particular,
+produce a valid, locked, or decided value) need to be eventually received by
+all correct processes.
+This opens up the possibility of implementing the `broadcast` primitive using a
+best-effort protocol, that does not guarantee Property 2, while resorting on
+the generically called [synchronization protocols][synchronization-spec] to
+ensure Property 2 for particular subsets of messages required to get out from
+deadlocks and ensure liveness.
 
-> TODO: most timeouts can be cancelled when the associated conditions are not
-> any longer observed (round or height changed, round step changed).
+### Timeouts
+
+The `schedule` primitive is adopted in the pseudo-code to schedule the
+execution of `OnTimeout<Step>(height, round)` functions to after the duration
+returned by `timeout<Step>(round)`,
+where `<Step>` is one of `Propose`, `Prevote`, and `Precommit` - the three
+[round steps](#round-steps).
+
+The `OnTimeout<Step>` functions define the actions taken upon an unsuccessful
+corresponding round step: the process votes `nil` or abandons the current round.
+These functions are guarded by conditions that verify if the state, in
+particular the round step, has not changed since when their execution was
+scheduled.
+It the state has changed, the execution may not produce any action.
+
+> Implementations could cancel scheduled timeouts when the state changes and
+> the associated conditions are no longer observed.
```

Comment:
I guess also when the process goes to the next round or if it decides and goes to the next height.

Reply:
I am explicitly citing the guard conditions, that are not detailed but explained in the previous paragraph.